frontend/src/hooks/queryBuilder/useGetQueryRange.ts

```diff
@@ -11,9 +11,14 @@ import { MetricRangePayloadProps } from 'types/api/metrics/getQueryRange';
 type UseGetQueryRange = (
 	requestData: GetQueryResultsProps,
 	options?: UseQueryOptions<SuccessResponse<MetricRangePayloadProps>, Error>,
+	isDependedDataLoaded?: boolean,
 ) => UseQueryResult<SuccessResponse<MetricRangePayloadProps>, Error>;
 
-export const useGetQueryRange: UseGetQueryRange = (requestData, options) => {
+export const useGetQueryRange: UseGetQueryRange = (
+	requestData,
+	options,
+	isDependedDataLoaded = false,
```

Comment:
also can we use `isQueryEnabled` seems naming is action driven rather than purpose driven

Reply:
Okay